pkg/controller/storagecluster/reconcile.go

```diff
@@ -1054,6 +1068,12 @@ func newStorageClassDeviceSets(sc *ocsv1.StorageCluster) []rook.StorageClassDevi
 				Portable:             ds.Portable,
 				TuneSlowDeviceClass:  ds.Config.TuneSlowDeviceClass,
 			}
+
+			err := validateMetadataPvc(sc)
+			if err == nil {
+				ds.MetadataPVCTemplate.ObjectMeta.Name = metadataPVCName
+				set.VolumeClaimTemplates = []corev1.PersistentVolumeClaim{ds.DataPVCTemplate, ds.MetadataPVCTemplate}
+			}
```

Comment:
What happens when `err != nil`? Shouldn't we be handling that?

Reply:
`err != nil `means the MetadataPVCTemplate is not defined, 

Since MetadataPVCTemplate is optional , no modification is required in newCephCluster when MetadataPVCTemplate not defined, so handling wouldn't be required.

MetadataPVCTemplate is added to the volumeclaim list only when it is defined ( when `err == nil` )

Reply:
I don't think that's a good idea. When an admin creates a StorageDeviceSet like this, he expects to have a metadata device. To simply ignore it and create the OSDs without it is bad since they can't just go back and fix it by changing the CR. We should return the error and refuse to continue reconciling.

Reply:
Correct me if i am wrong,

- if we return error and refuse to reconcile when metadatapvc is not defined, then providing the MetadataPVCTemplate in the storage cluster yaml becomes a mandatory thing. (i.e) user has to provide both DataPVCTemplate and MetadataPVCTemplate.

- Based on my understanding, the MetadataPVCTemplate is optional and the user can create the storage cluster even without providing the MetadataPVCTemplate in storage cluster yaml

Reply:
I would imagine you're missing a check for `ds.MetadataPVCTemplate != nil` or something.

Reply:
Thanks, Fixed .
Implemented the template as a pointer for null check